packages/iris-grid/src/IrisGridTreeTableModel.ts

```diff
@@ -44,47 +44,98 @@ class IrisGridTreeTableModel extends IrisGridTableModelTemplate<
-              constituentType: TableUtils.dataType.STRING,
-              isPartitionColumn: false,
-              isSortable: false,
-              isProxy: true,
-              description: 'Key column',
-              index: -1,
-              filter: () => {
-                throw new Error('Filter not implemented for virtual column');
-              },
-              sort: () => {
-                throw new Error('Sort not implemented virtual column');
-              },
-              formatColor: () => {
-                throw new Error('Color not implemented for virtual column');
-              },
-              get: () => {
-                throw new Error('get not implemented for virtual column');
-              },
-              getFormat: () => {
-                throw new Error('getFormat not implemented for virtual column');
-              },
-              formatNumber: () => {
-                throw new Error(
-                  'formatNumber not implemented for virtual column'
-                );
-              },
-              formatDate: () => {
-                throw new Error(
-                  'formatDate not implemented for virtual column'
-                );
-              },
+    console.log(
+      'IrisGridTreeTableModel constructor',
+      this.formatter.showExtraGroupColumn
+    );
+    this.virtualColumns = this.formatter.showExtraGroupColumn
+      ? [
+          {
+            name: '__DH_UI_GROUP__',
+            displayName: 'Group',
+            type: TableUtils.dataType.STRING,
+            constituentType: TableUtils.dataType.STRING,
+            isPartitionColumn: false,
+            isSortable: false,
+            isProxy: true,
+            description: 'Key column',
+            index: -1,
+            filter: () => {
+              throw new Error('Filter not implemented for virtual column');
+            },
+            sort: () => {
+              throw new Error('Sort not implemented virtual column');
+            },
+            formatColor: () => {
+              throw new Error('Color not implemented for virtual column');
+            },
+            get: () => {
+              throw new Error('get not implemented for virtual column');
+            },
+            getFormat: () => {
+              throw new Error('getFormat not implemented for virtual column');
+            },
+            formatNumber: () => {
+              throw new Error(
+                'formatNumber not implemented for virtual column'
+              );
+            },
+            formatDate: () => {
+              throw new Error('formatDate not implemented for virtual column');
+            },
+          },
+        ]
+      : [];
+  }
+
+  updateFormatterSetting(newFormatter: Formatter): void {
+    if (
+      this.formatter.showExtraGroupColumn !== newFormatter.showExtraGroupColumn
+    ) {
+      this.formatter.showExtraGroupColumn = newFormatter.showExtraGroupColumn;
+      this.updateVirtualColumns();
+    }
+  }
```

Comment:
This isn't being called from anywhere.
In `IrisGrid.tsx`, you're already creating a new formatter when the settings change, which will call `set formatter`. Override that method and call your `updateVirtualColumns` from there, e.g.:
```typescript
set formatter(newFormatter: Formatter) {
  const oldFormatter = super.formatter;
  super.formatter = newFormatter;
  if (newFormatter.showExtraGroupColumn !== oldFormatter.showExtraGroupColumn) {
    this.updateVirtualColumns();
  }
}
```